### ISSUE TITLE
[API Gateway] Impossible to getDomainPathMapping/updateBasePathMapping for basePath = '/'

### ISSUE THREAD

**Issue by taylorotwell** [aws/aws-sdk-php#1677]
It doesn't seem possible to perform these operations with a basePath of `/` or `''`. Attempting to updateBasePathMapping will result in: 

`Unable to determine service/operation name to be authorized`

`getBasePathMapping` with a basePath of `/` just seems to return some generic meta-data. Nothing related to the actual basePathMapping.

You are able to `createBasePathMapping` correctly.

**Comment by diehlaws** [comment]
Hello @taylorotwell, thanks for reaching out to us. While a bit confusing, this is expected behavior from the service's API. Per the [Get Base Path Mapping](https://docs.aws.amazon.com/apigateway/api-reference/link-relation/basepathmapping-by-base-path/) and [Update Base Path Mapping](https://docs.aws.amazon.com/apigateway/api-reference/link-relation/basepathmapping-update/) API reference pages:

> To specify an empty base path in the request URL, set the base_path path variable to (none).

I was able to reproduce the behavior you're describing when setting`'basePath'=> ''`, both for `getBasePathMapping` and `updateBasePathMapping` calls, however when setting `'basePath' => '(none)'` both operations worked as expected. 

Unfortunately this behavior is not made clear in the documentation for the AWS SDK for PHP (or our other SDKs), so I've reached out to the appropriate team internally to have this note included in our documentation.
